packages/payment-processor/src/payment/erc20-fee-proxy.ts

```
@@ -116,3 +156,48 @@
+ * @param amount optionally, the amount to pay. Defaults to remaining amount of the request.
+ * @param feeAmountOverride optionally, the fee amount to pay. Defaults to the fee amount of the request.
+ */
+export function preparePrivateErc20FeeProxyPaymentTransaction(
+  request: ClientTypes.IRequestData,
+  amount?: BigNumberish,
+  feeAmountOverride?: BigNumberish,
+): IPreparedPrivateTransaction {
+  validateErc20FeeProxyRequest(request, amount, feeAmountOverride);
+
+  const { value: tokenAddress, network } = request.currencyInfo;
+  EvmChains.assertChainSupported(network!);
```

Biome

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)